crates/ruff/src/rules/pyflakes/rules/undefined_local.rs

```diff
@@ -18,26 +18,41 @@ impl Violation for UndefinedLocal {
     }
 }
 
-/// F821
-pub fn undefined_local(name: &str, scopes: &[&Scope], bindings: &Bindings) -> Option<Diagnostic> {
-    let current = &scopes.last().expect("No current scope found");
-    if matches!(current.kind, ScopeKind::Function(_)) && !current.defines(name) {
-        for scope in scopes.iter().rev().skip(1) {
-            if matches!(scope.kind, ScopeKind::Function(_) | ScopeKind::Module) {
-                if let Some(binding) = scope.get(name).map(|index| &bindings[*index]) {
-                    if let Some((scope_id, location)) = binding.runtime_usage {
-                        if scope_id == current.id {
-                            return Some(Diagnostic::new(
-                                UndefinedLocal {
-                                    name: name.to_string(),
-                                },
-                                location,
-                            ));
-                        }
-                    }
+/// F823
+pub fn undefined_local(checker: &mut Checker, name: &str) {
```

Comment:
Refactored this a bit, it should also be more performant (previously, we created a `Vec` of `Scope` to pass in every time).